enclave/sgx/attester.c

```diff
@@ -365,16 +365,22 @@ static oe_result_t _get_attester_plugins(
     // It's possible that there is no supported format
     if (temporary_buffer_size >= sizeof(oe_uuid_t))
     {
+        size_t allocated_buffer_size = temporary_buffer_size;
         // Allocate buffer to held the format IDs
         temporary_buffer = (uint8_t*)oe_malloc(temporary_buffer_size);
         if (temporary_buffer == NULL)
             OE_RAISE(OE_OUT_OF_MEMORY);
 
         // Get the format IDs
         result = oe_get_supported_attester_format_ids_ocall(
-            (uint32_t*)&retval, temporary_buffer, temporary_buffer_size, NULL);
+            (uint32_t*)&retval,
+            temporary_buffer,
+            allocated_buffer_size,
+            &temporary_buffer_size);
         OE_CHECK(result);
         OE_CHECK(retval);
+        if (temporary_buffer_size > allocated_buffer_size)
+            OE_RAISE(OE_UNEXPECTED);
```

Comment:
Because this spans an OCALL, we've been pretty strict in checking that the returned size is exactly equal the originally specified size (i.e. also should not be smaller.)

Reply:
Agreed we can check to ensure the two values are the same. Will update the code.